frontend/lib/features/pfis/pfi_providers.dart

```diff
@@ -1,22 +1,56 @@
+import 'dart:convert';
 import 'package:flutter_starter/features/pfis/pfi.dart';
 import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:http/http.dart' as http;
+import 'package:shared_preferences/shared_preferences.dart';
 
-final pfisProvider = Provider<List<Pfi>>(
-  (ref) => [
-    Pfi(
-      id: 'prototype',
-      name: 'Prototype',
-      didUri: 'did:dht:74hg1efatndi8enx3e4z6c4u8ieh1xfkyay4ntg4dg1w6risu35y',
-    ),
-    Pfi(
-      id: 'africa',
-      name: 'Africa',
-      didUri: 'coming soon...',
-    ),
-    Pfi(
-      id: 'mexico',
-      name: 'Mexico',
-      didUri: 'coming soon...',
-    ),
-  ],
-);
+final pfisProvider = FutureProvider<List<Pfi>>((ref) async {
+  const url = 'https://raw.githubusercontent.com/TBD54566975/pfi-providers-data/main/pfis.json';
+  const cacheKey = 'pfis_cache';
```

Comment:
super nit: We can make this private to the file to avoid having to pass it to the `_loadFromCache` function below.